webviz_subsurface/plugins/_group_tree/group_tree_data.py

```diff
@@ -0,0 +1,478 @@
+from typing import List, Dict, Tuple, Any
+import logging
+
+import pandas as pd
+import numpy as np
+
+from webviz_config.common_cache import CACHE
+
+
+class GroupTreeData:
+    """This class holds the summary and gruptree datasets and functionality
+    to combine the data and calculate the GroupTree component input dataset.
+    """
+
+    def __init__(self, smry: pd.DataFrame, gruptree: pd.DataFrame):
+        self.smry = smry
+        self.gruptree = gruptree
+        self.ensembles = gruptree["ENSEMBLE"].unique()
```

Comment:
Could it be nice to include typing information here?

ala `self.ensembles : Set[str] = gruptree["ENSEMBLE"].unique()`